src/Raven.Server/Documents/Replication/ReplicationItems/ReplicationBatchItem.cs

```diff
@@ -24,6 +25,23 @@ public abstract class ReplicationBatchItem : IDisposable
         protected Reader Reader;
 
         private List<IDisposable> _garbage;
+        private readonly DisposeOnce<SingleAttempt> _disposeOnce;
+
+        protected ReplicationBatchItem()
+        {
+            _disposeOnce = new DisposeOnce<SingleAttempt>(() =>
```

Comment:
it might be too heavy for this flow, better to just use `bool` - no concurrency concerns here

Reply:
Done